tests/tests.rs

```diff
@@ -229,50 +230,80 @@ async fn test_on_disk_index() -> Result<()> {
     common::clean(storage, path).await
 }
 
-#[tokio::test]
-async fn test_work_dir_lock() {
-    use nix::sys::wait::{waitpid, WaitStatus};
-    use nix::unistd::{fork, ForkResult};
-
-    let path = common::init("work_dir_lock");
-
+#[test]
+#[named]
+fn test_work_dir_lock() {
+    use rusty_fork::{fork, rusty_fork_id};
+    use std::fs::{create_dir_all, read, write};
+    use std::panic;
+    use std::path::Path;
+    use std::sync::Arc;
+    use tokio::runtime::Builder;
+
+    const MAX_CHILD_WAIT_COUNT: usize = 30;
+    const MAX_PARENT_WAIT_SECONDS: u64 = 120;
+
+    let path = Arc::new(common::init(function_name!()));
+    let init_file = Path::new(path.as_ref()).join("main_init_ready.special");
+    let child_init_file = init_file.clone();
+    let child_path = path.clone();
     // We need separate processes because locks do not work within the same process
-    match unsafe { fork() } {
-        Ok(ForkResult::Parent { child }) => {
+    fork(
+        function_name!(),
```

Comment:
You should keep `rusty_fork_test_name!()` too, because according to the source code it adds the module name (https://docs.rs/rusty-fork/latest/src/rusty_fork/fork_test.rs.html#121-126):
 ```suggestion
        rusty_fork_test_name!(function_name!()),
```